### DISCUSSION TITLE
[QUESTION] Can longhorn be used across clusters

### DISCUSSION THREAD

**Discussion by wangch1030** [longhorn/longhorn#8211] [General]
## Question

 - The Longhorn deployed in Harvester is designed to work within the cluster, making it difficult to identify disks outside the cluster.
 - Reference link: https://github.com/longhorn/longhorn/issues/2192
 - Have any changes been made to cross cluster usage of longhorn within the Harvester cluster at this stage

**Comment by ejweber** [comment]
Converted to a discussion.

Longhorn has not taken any steps towards usage across clusters since the linked issue. It is still the Longhorn design philosophy that it provides highly available storage WITHIN a cluster (though features like [Disaster Recovery Volumes](https://longhorn.io/docs/1.6.0/snapshots-and-backups/setup-disaster-recovery-volumes/) can be used to allow a secondary cluster to quickly takeover in the event of the failure of a primary cluster.

For downstream clusters within a Harvester cluster, I think the Harvester team generally intends for users to install the [Harvester CSI driver](https://docs.harvesterhci.io/v1.2/rancher/csi-driver) backed by a Longhorn storage class in the Harvester cluster itself. This allows Longhorn to operate at (and manage redundancy at) the level of the physical disk.
